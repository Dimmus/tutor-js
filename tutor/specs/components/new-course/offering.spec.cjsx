{React, SnapShot} = require '../helpers/component-testing'


find = require 'lodash/find'
cloneDeep = require 'lodash/cloneDeep'

OfferingTitle = require '../../../src/components/new-course/offering-title'
OFFERINGS = require '../../../api/offerings'

{OfferingsStore, OfferingsActions} = require '../../../src/flux/offerings'
{NewCourseStore, NewCourseActions} = require '../../../src/flux/new-course'


describe 'CreateCourse: choosing offering', ->

  beforeEach ->
    OfferingsActions.loaded(OFFERINGS)
    @props =
      offeringId: OFFERINGS.items[0].id

  it 'doesnâ€™t blow up when appearance code from offering is invalid', ->
    burntOfferings = cloneDeep(OFFERINGS)
    find(burntOfferings.items, id: @props.offeringId).appearance_code = 'firefirefire'
    OfferingsActions.loaded(burntOfferings)
    wrapper = shallow(<OfferingTitle {...@props} />)
    expect(wrapper.is('[data-appearance="firefirefire"]')).to.be.true
    undefined

  # snapshots currently do not work because of bug with React 15.3.x.
  # is fixed in 15.4 https://github.com/facebook/react/issues/7386
  it 'matches snapshot', ->
    component = SnapShot.create(
      <OfferingTitle {...@props} />
    )
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()
