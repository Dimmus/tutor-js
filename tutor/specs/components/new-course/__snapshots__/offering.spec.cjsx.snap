// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCourse: choosing offering matches snapshot 1`] = `
<div
  className="course-offering"
  data-appearance="college_biology"
>
  <div
    className="contents"
  >
    <div
      className="title"
    >
      College Biology
    </div>
    <div
      className="sub-title"
    />
  </div>
</div>
`;
